libs/contract/src/lib/avails/avails.ts

```diff
@@ -203,7 +203,11 @@ function isAvailInTerm<T extends BucketTerm | Term>(avail: MapAvailsFilter, term
 }
 
 function getMatchingMapMandates(mandates: FullMandate[], avails: MapAvailsFilter): FullMandate[] {
-  return mandates.filter(mandate => mandate.terms.some(term => isMapTermInAvails(term, avails)));
+  const filteredMandates= mandates.filter(mandate => mandate.terms.some(term => isMapTermInAvails(term, avails)));
+  return filteredMandates.map(({terms, ...rest}) => ({
+    ...rest,
+    terms:terms.filter(term => isMapTermInAvails(term, avails))
+  }));
```

Comment:
Do it the other way around : 
```suggestion
  return mandates
    .map(mandate => {
      mandate.terms = mandate.terms.filter(term => isMapTermInAvails(term, avails));
      return mandate;
    })
    .filter(mandate => mandate.terms.length);
```